examples/kafka/dev-values-tls-all-components-and-kafka-and-oauth2-low-resource.yaml

```diff
@@ -0,0 +1,320 @@
+
+cert-manager:
+  enabled: true
+
+createCertificates:
+  selfSignedPerComponent:
+    enabled: true
+
+openid:
+  enabled: true
+  # From token generated in Okta UI or other method:
+  issuerUrl: https://dev-42..16.okta.com/oauth2/au...15d7
+  scope: pulsar_client_m2m
+  client_id: "0oa7...nDd5d7"
+  client_secret: "CL08ZNh...tF2bP"
+  withS4k: true
+zookeeper:
+  replicaCount: 1
+  resources:
+    requests:
+      memory: 300Mi
+      cpu: 0.2
+  configData:
+    PULSAR_MEM: "-Xms300m -Xmx300m -Djute.maxbuffer=10485760 -XX:+ExitOnOutOfMemoryError"
+
+bookkeeper:
+  replicaCount: 1
+  resources:
+    requests:
+      memory: 356Mi
+      cpu: 0.1
+  configData:
+    BOOKIE_MEM: "-Xms156m -Xmx156m -XX:MaxDirectMemorySize=200m -XX:+ExitOnOutOfMemoryError"
+
+brokerSts:
+  component: broker
+  replicaCount: 1
+  service:
+    headless: true
+  ledger:
+    defaultEnsembleSize: 1
+    defaultAckQuorum: 1
+    defaultWriteQuorum: 1
+  resources:
+    requests:
+      memory: 300Mi
+      cpu: 0.3
+  configData:
+    PULSAR_MEM: "-Xms200m -Xmx200m -XX:MaxDirectMemorySize=100m -XX:+ExitOnOutOfMemoryError"
+    PULSAR_EXTRA_OPTS: "-Dpulsar.log.root.level=info"
+
+autoRecovery:
+  enableProvisionContainer: true
+  resources:
+    requests:
+      memory: 300Mi
+      cpu: 0.1
+
+function:
+  replicaCount: 1
+  functionReplicaCount: 1
+  resources:
+    requests:
+      memory: 256Mi
+      cpu: 0.1
+  authenticationProviders: "com.datastax.oss.pulsar.auth.AuthenticationProviderOpenID,org.apache.pulsar.broker.authentication.AuthenticationProviderToken"
+  configData:
+    PULSAR_MEM: "-Xms156m -Xmx156m -XX:MaxDirectMemorySize=100m -XX:+ExitOnOutOfMemoryError"
+    PULSAR_EXTRA_OPTS: "-Dpulsar.log.root.level=info"
+
+proxy:
+  replicaCount: 1
+  resources:
+    requests:
+      memory: 256Mi
+      cpu: 0.1
+  wsResources:
+    requests:
+      memory: 256Mi
+      cpu: 0.1
+  authenticationProviders: "org.apache.pulsar.broker.authentication.AuthenticationProviderToken"
+  wsAuthenticationProviders: "com.datastax.oss.pulsar.auth.AuthenticationProviderOpenID,org.apache.pulsar.broker.authentication.AuthenticationProviderToken"
+  configData:
+    PULSAR_MEM: "-Xms200m -Xmx200m -XX:MaxDirectMemorySize=56m"
+    PULSAR_EXTRA_OPTS: "-Dpulsar.log.root.level=info"
+    PULSAR_PREFIX_kafkaListeners: "SASL_PLAINTEXT://0.0.0.0:9092,SASL_SSL://0.0.0.0:9093"
+    # here you have to customize the advertised name if you want to access
+    # Kafka from outside the kube    
+    # Drop the SASL_PLAINTEXT endpoint if you want to allow only TLS (here and above)
+    # The Proxy uses the same TLS certificate configured for the Pulsar protocol
+    #    PULSAR_PREFIX_kafkaAdvertisedListeners: "SASL_PLAINTEXT://pulsar-proxy:9092,SASL_SSL://pulsar-proxy:9093"
+    # if you use port forwarding to localhost then you have to set 'localhost'    
+    PULSAR_PREFIX_kafkaAdvertisedListeners: "SASL_PLAINTEXT://pulsar-proxy:9092,SASL_SSL://pulsar-proxy:9093"
```

Comment:
@michaeljmarshall I want to find a way to abstract all this Kafka-specific configuration, as well as the Kafka configs in the proxy extensions section, into something that starts like:


```
starlight:
  kafka:
    enabled: true
    ssl: true
    . . .
```
However, it's not clear to me which of these parameters a customer would likely need to override. (How often do people really override these ports at this level instead of just using a load balancer to do that?)